### ISSUE TITLE
Nonfunctional in Thunderbird 60.0b9 (32-bit) Beta

### ISSUE THREAD

**Issue by DNIU14** [protz/Manually-Sort-Folders#64]
Very disappointing that it is totally nonfunctional for Thunderbird 60.0b9 (32-bit) Beta.

**Comment by monpopoki** [comment]
Iaorana !

I confirme it' dosen't works. When i open the windows to sort, the accounts are not extracted:

![image](https://user-images.githubusercontent.com/40663874/43245286-09eb869c-90ae-11e8-86fd-2bf0ecd97192.png)

Thanks
François

**Comment by schneeschmelze** [comment]
Same here on macOS 10.13.6. :(

**Comment by Berenke** [comment]
Same error on TB60.0 / Win8.1 64bit.

I have two remarks.

First there isn't the latest version on the AMO site. If I see right, the latest changes occurred about 3 months ago but on AMO site there is a version from 1st of May 2016 (I checked this in the code, there is the same situation).

Second I made a little test with the latest addon version. I saved the oldest TB before install the new, so I can get the addon's folder order settings from it. Then I installed TB 60 and addon, set the folder order to manually defined and put the old settings with Preferences > Advanced > Config Editor. Aftrer it I restarted TB and folders order was I got used to it. Then I checked the folder list in addon's settings, it was empty. When I close the addon's window the TB's folder order is messed up. I'm not a programmer but because of the above I think this is a settings saving or settings viewing error in the addon.

Beren

(Sorry for my bad English, this isn't my native language but I hope there is understandable.)

**Comment by GH-User** [comment]
Same here on TB60.0 32bit, Win7 64bit.

**Comment by gjf** [comment]
Same on TB60.0 64 bit, Win 10 Pro x64.

**Comment by yan12125** [comment]
TL;DR - @protz Please release a new version for Thunderbird 60 compatibility.

---

Below are my findings. First, there's a line in my terminal:
```
JavaScript error: chrome://tbsortfolders/content/folderPane.js, line 56: SyntaxError: missing { before function body
```
The line 56 of the mentioned file in `extensions/tbsortfolders@xulforum.org.xpi` is:
```JavaScript
        tbsf_prefs_functions[key] = function (a,b) tbsf_sort_functions[2](tbsf_data[key][1], a, b);
```
This line has been fixed in https://github.com/protz/Manually-Sort-Folders/commit/88cebfa4060ce4dfc8a12e36d81ec4114cc349e7. Above all, a new version should fix the issue.

**Comment by TCHessen** [comment]
With this fixes in https://github.com/protz/Manually-Sort-Folders/commit/88cebfa4060ce4dfc8a12e36d81ec4114cc349e7 the extension works a bit more better, but the folder pane does not load. Maybe I find some time to fix it, but I cannot update the repository.

**Comment by GH-User** [comment]
I can verify @TCHessen  that it is still not working with this fix.

**Comment by biggybean** [comment]
Backing this - Win10 Home 64bit, TB 60.0 32bit

**Comment by Berenke** [comment]
@TCHessen: I tried to change "builderVew" to "builderView" or simple "builder" - I thought it's just a spelling mistake. Not fixed anything but made more errors.

**Comment by TCHessen** [comment]
The main problem is that (in ui.js:198) the entity folders_tree.builderView (or folders_tree.builder) is null, but in the prior version of tb not. So there are major changes in the api that need more time than just to fix a typo.

**Comment by GH-User** [comment]
Hopefully someone will find a solution because it seems that the original author doesn't work any more on this very useful addon.

**Comment by rezeptpflichtig** [comment]
> The main problem is that (in ui.js:198) the entity folders_tree.builderView (or folders_tree.builder) is null, but in the prior version of tb not. So there are major changes in the api that need more time than just to fix a typo.

pretty sure it should be folders_tree.builder
but the folders_tree is already null.
in ui.js:190 it is calling document.getElementByID("foldersTree")
but there is no such ID in the DOM.. but a folderTree
however changing that did not help either and I don't know how the folder list is eaxtly build..

**Comment by Othneios** [comment]
in the ui debugger there is an element with ID "folderTree". You got the missing "s"

**Comment by hb9tst** [comment]
@rezeptpflichtig Indeed it doesn't help, the big issue is the removal of RDF-based trees in XUL (see comments on issue #67 for more details), so there is no easy fix and that part of the UI has to be reimplemented.

**Comment by x-ji** [comment]
Duplicate of #67? The version by @Ximi1970 at https://github.com/Ximi1970/Manually-Sort-Folders/releases seems to work.

**Comment by Gluups** [comment]
Hello, compared to the version that is (still) installed by the extensions manager, Ximi1970's version has the advantage in Thunderbird 60 to display the names of the accounts and of the folders.
Nevertheless I see no up/down buttons. It is right that this problem is mentioned in bug #67, but is there any solution ?

**Comment by Ximi1970** [comment]
Change the "Use Thunderbird's default" to "Use manually defined sort function"...

**Comment by Gluups** [comment]
Sorry about my previous message : I did not select the manual order, so it was normal there were no up/down buttons.
So, after migrating to Thunderbird 60, I did not test modifying the orders, but the aspect of the user interface seems correct.
Well, when everything is good, it can be a good idea to do a proper deployment, so that people get the correct version through the extensions manager.
I found a good version because I followed this thread.

**Comment by Gluups** [comment]
Sorry not to have been quick enough to say I found the reason ...

**Comment by protz** [comment]
My sincerest apologies for the long delay with this update. Various life things got in the way, not to mention my research work which is my main focus of interest these days. I am happy to announce that, in spite of it all, this addon which I created while an undergrad nearly 10 years ago just received an update that makes it compatible with Thunderbird 60! Thunderbird should update automatically in a few days, as soon as the new version is accepted. If you'd rather upgrade now:
- download the version "1.2.0pre3" listed in the comments above, or from my website: https://jonathan.protzenko.fr/manually-sort-folders/manually-sort-folders-201812152042.xpi
- then, in Tools > Addons, hit the cog, the do "install addon from file".

Thanks again to @Ximi1970 for the update.

Hope this helps!

Best,

Jonathan
